sr5824-msr9732-sm9453/project.md

```diff
@@ -0,0 +1,136 @@
+---
+title: "Interoperability in the Heterogeneous Cloud Environment"
+author:
+  - Sriram Ramesh <sriram.ramesh@nyu.edu>
+  - Metarya Ruparel <msr9732@nyu.edu>
+  - Safwan Mahmood <sm9453@nyu.edu>
+---
+
+For our Big Data & Machine Learning final project we plan to hypothesize a solution
+for **Interoperability in the Heterogeneous Cloud Environment**, a blatant issue 
+industry wide.
+
+# Motivation
+
+After the burst of cloud platforms in 2006 with the introduction of Amazon
+Web Services, a lot of companies started migrating their services to the 
+cloud platform from a bare metal lying in one of their own datacenters.
+Big hulking datacenters are a necessary evil, a barely concealed secret. 
+No one wants to build one. They’re certainly impressive – they resemble the 
+set of a science fiction movie, row upon row of silent machines. But they’re 
+wildly expensive, and their maintenance challenges are never ending. 
+
+Cloud Platforms provided these companies the ability to access shared, online computing 
+resources, without the over head of maintaining them. However, these providers often 
+offer their own proprietary applications, interfaces, APIs and infrastructures, which
+somewhat makes these providers unique. Due to the uniqueness (be it performance or costing)
+companies would want to onboard different set of services onto various providers.
+For example, Salesforce.com, which  started the wave of SaaS back in the late 1900's 
+, has only recently started it's migration from 1st party datacenters to these 3rd party
+providers. Some of their micro-services are being on-boarded to GCP, while a couple of
+others to AWS, resulting in a heterogeneous cloud environment. This heterogeneous 
+environment makes it difficult for companies to change cloud service providers; in 
+any case. Exploring capabilities to support the automated migration from one provider 
+to another is an active, open research area. A lot of people have been pursuing approaches 
+to reduce the impact of vendor lock-in by investigating the cloud migration problem 
+at the level of the VM. However, the migration downtime, decoupling VM from
+underlying systems and security of live channels remain open issues and as a part of
```

Comment:
No idea what "security of live channels" means here or what problem you are highlighting.

Reply:
When migrating from one provider to another, you would actually be moving a lot of stuff, your DB (in some cases this might not be necessary), your file system (places where you store .secrets, or any batch job that you have scheduled, your cron jobs basically). When moving these things over a channel from your current environment to another, you could be prone to attacks, if we have no encryption on this open channel. There could be someone snooping, who could read your .secrets or change your scheduled batch jobs. We are not targetting this as a part of this proposal but was mentioning one of the many issues that an individual can encounter when migrating. We could actually remove this part from the proposal to make the proposal more succinct.